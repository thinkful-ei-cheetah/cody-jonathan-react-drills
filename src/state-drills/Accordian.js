import React, { Component } from 'react';

class Accordian extends Component {
    
    constructor(props) {
        super(props)
      }
    
    state = {
       sections : this.props.sections,
       activeIndex: null
    }
    handleClickedInfo = (index) => {
        this.setState({
        activeIndex: index
        })
        console.log(index)
    }

    render(){
        return(
           <ul>

            {this.state.sections.map((section,index) => 
            <li key = {index}>
                <button onClick = {() => this.handleClickedInfo(index)}>{section.title}</button> 
                <p>{this.state.activeIndex === index ? section.content:null}</p>
            </li>)}
               
           </ul>
       );
   }
}
export default Accordian