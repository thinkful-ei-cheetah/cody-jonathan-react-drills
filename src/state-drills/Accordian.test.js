import React from 'react';
import ReactDOM from 'react-dom';
import Accordian from './Accordian';
import renderer from 'react-test-renderer';
const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]
  describe('Stage Participants', () => {
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Accordian sections={sections}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});
 it('renders UI as expected', () => {
        const tree = renderer
            .create(<Accordian sections={sections}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})
