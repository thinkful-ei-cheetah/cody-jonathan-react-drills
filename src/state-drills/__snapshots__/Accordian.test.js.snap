// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stage Participants renders UI as expected 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p />
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p />
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p />
  </li>
</ul>
`;
